crates/next-dev/benches/mod.rs

```diff
@@ -230,21 +182,32 @@ fn bench_hmr_internal(mut g: BenchmarkGroup<WallTime>, location: CodeLocation) {
                                         "Unable to evaluate JavaScript in the page for HMR check \
                                          flag",
                                     )?;
-                                Ok(guard)
+
+                                // TODO(alexkirsz) Turbopack takes a few ms to start listening on
+                                // HMR, and we don't send updates retroactively, so we need to wait
+                                // before starting to make changes.
+                                // This should not be required.
+                                tokio::time::sleep(Duration::from_millis(5000)).await;
+
+                                Ok((modules, guard))
                             },
-                            |mut guard| async move {
-                                // Make 5 changes to warm up.
-                                for _ in 0..5 {
-                                    let _ =
-                                        make_change(&mut guard, location, MAX_UPDATE_TIMEOUT).await;
+                            |(modules, guard)| async move { Ok((modules, guard)) },
+                            |measurement, (mut modules, mut guard)| async move {
+                                let mut value = measurement.zero();
+                                let measurement_count = modules.len() as u32;
+                                while let Some(module) = modules.pop() {
+                                    let change_duration = make_change(
+                                        &module,
+                                        &mut guard,
+                                        location,
+                                        MAX_UPDATE_TIMEOUT,
+                                        &measurement,
+                                    )
+                                    .await?;
+                                    value = measurement.add(&value, &change_duration);
                                 }
-                                Ok(guard)
-                            },
-                            |mut guard| async move {
-                                make_change(&mut guard, location, MAX_UPDATE_TIMEOUT).await?;
-
                                 // Defer the dropping of the guard to `teardown`.
-                                Ok(guard)
+                                Ok((value / measurement_count, guard))
```

Comment:
Reporting an average only breaks the statistics computed by criterion. e.g. the std dev is incorrect.

Reply:
Agreed (see PR description). We should have https://github.com/vercel/turbo/pull/2698 alongside the other changes in this PR. I can work on separating the changes.